keras/engine/training_utils.py

```diff
@@ -417,6 +417,14 @@ def weighted(y_true, y_pred, weights, mask=None):
             weight_ndim = K.ndim(weights)
             score_array = K.mean(score_array,
                                  axis=list(range(weight_ndim, ndim)))
+            # reduce weight array to same ndim as score_array (needed for
+            # sample_weight_mode='element') by checking that any extra
+            # dimensions in weights are size 1, i.e. dummy dimensions
+            ndim = K.ndim(score_array)
+            if weight_ndim > ndim and np.any(K.shape(weights)[ndim:] == 1):
+                raise ValueError('weights array shape is incompatible with '
+                                 'loss function')
```

Comment:
This function is used for metrics as well so may be we need to rephrase this?

Reply:
With the code simplifcation above, this is no longer an issue